src/core/response/parameters.rs

```diff
@@ -26,18 +28,26 @@ impl From<IdToken> for Json {
     }
 }
 
-// TODO: Update this type to something like:
-//
-// enum VpToken {
-//     Single(String),
-//     SingleAsMap(Map<String, Value>),
-//     Many(Vec<VpToken>),
-// }
-//
-// See: https://github.com/spruceid/oid4vp-rs/pull/8#discussion_r1750274969
-//
+/// OpenID Connect for Verifiable Presentations specification defines `vp_token` parameter:
+///
+/// > JSON String or JSON object that MUST contain a single Verifiable Presentation or
+/// > an array of JSON Strings and JSON objects each of them containing a Verifiable Presentations.
+/// >
+/// > Each Verifiable Presentation MUST be represented as a JSON string (that is a Base64url encoded value)
+/// > or a JSON object depending on a format as defined in Appendix A of [OpenID.VCI].
+/// >
+/// > If Appendix A of [OpenID.VCI] defines a rule for encoding the respective Credential
+/// > format in the Credential Response, this rules MUST also be followed when encoding Credentials of
+/// > this format in the vp_token response parameter. Otherwise, this specification does not require
+/// > any additional encoding when a Credential format is already represented as a JSON object or a JSON string.
+///
+/// See: [OpenID.VP#section-6.1-2.2](https://openid.net/specs/openid-4-verifiable-presentations-1_0.html#section-6.1-2.2)
 #[derive(Debug, Clone)]
-pub struct VpToken(pub String);
+pub enum VpToken {
+    Single(String),
+    SingleAsMap(Json),
```

Comment:
Better to represent as a `Map<String, Json>` to avoid having to unwrap the `Json::Object`